### PR TITLE
model_collect: add more services

### PR DESCRIPTION
**Pull request by Morriar**
Just some services that work on flattened views of the model that I will need soon.

### CODE REVIEW

File: src/model/model_collect.nit [hunk 1]
```diff
@@ -33,6 +33,61 @@ import model_views
...
+
+	# Build the importation poset for `self`
+	fun importation_poset(view: ModelView): POSet[MModule] do
+		var mmodules = new HashSet[MModule]
+		mmodules.add self
+		mmodules.add_all collect_ancestors(view)
+		mmodules.add_all collect_parents(view)
+		mmodules.add_all collect_children(view)
+		mmodules.add_all collect_descendants(view)
+		return view.mmodules_poset(mmodules)
```
Comment (privat) [inline src/model/model_collect.nit]: I'm not sure what `mmodules_poset` is

**Comment by privat** [comment]
+1

**Comment by Morriar** [comment]
Reroll, I missed the first commit

**Comment by privat** [comment]
still +1

**Comment by privat** [comment]
Maybe you could use the `sub` method proposed in #2029 to simplify mmodules_poset and mclasses_poset.

~~~nit
fun mmodules_poset(mmodules: Set[MModule]) do
   return model.module_importation_hierarchy.sub(mmodules)
end

fun mclasses_poset(mainmodule: MModule, mclasses: Set[MClass]): POSet[MClass] do
   return mainmodule.flatten_class_hierarchy.sub(mclasses)
end
~~~

**Comment by Morriar** [comment]
Reroll rebased on top of #2029, @privat this is on your hands!

**Pull request merged**